draft-ietf-httpbis-http2bis.xml

```diff
@@ -2587,23 +2587,23 @@
           <name>HTTP Fields</name>
           <t>
             HTTP fields carry information as a series of field lines, which are key-value pairs. For a listing of
-            registered HTTP fields, see the "Hypertext Transfer Protocol (HTTP) Field Name Registry" registry maintained at <eref target="https://www.iana.org/assignments/http-fields/"/>.
+            registered HTTP fields, see the "Hypertext Transfer Protocol (HTTP) Field Name Registry"
+            registry maintained at <eref target="https://www.iana.org/assignments/http-fields/"/>.
           </t>
           <t>
             Field names are strings of ASCII characters that are compared in a case-insensitive
             fashion. Field names MUST be converted to lowercase when constructing a HTTP/2
-            message. A request or response containing an uppercase character ('A' to 'Z', ASCII 0x41
-            to 0x5a) in a field name MUST be treated as <xref target="malformed">malformed</xref>.
+            message.
           </t>
           <t>
             HPACK is capable of carrying field names or values that are not valid in HTTP.  Though
             HPACK can carry any octet, fields are not valid in the following cases:
           </t>
           <ul>
             <li>
-              A field name MUST NOT contain characters in the range 0x00-0x20 or 0x7F-0xFF (both
-              ranges inclusive).  This limits field names to visible ASCII characters, other than
-              ASCII SP (0x20).
+              A field name MUST NOT contain characters in the range 0x00-0x20, 0x41-0x5A or 0x7F-0xFF
+              (all ranges inclusive).  This limits field names to visible ASCII characters, excluding
+              ASCII SP (0x20) and uppercase characters ('A' to 'Z', ASCII 0x41 to 0x5a).
```

Comment:
@mnot to get rid of the gap problem, we could just increase the minimal validation to exclude all field names that 
break HTTP syntax (but not semantics).
```suggestion
              A field name MUST NOT contain characters 0x00-0x20, 0x22, 0x28, 0x29,
              0x2C, 0x41-0x5D, 0x7B, 0x7D or 0x7F-0xFF (all ranges inclusive).
              This limits field names to visible ASCII characters permitted by the HTTP ABNF 
              grammar from <xref target="HTTP" section="5"/>, other than uppercase characters 
              ('A' to 'Z', ASCII 0x41 to 0x5a).
```
The down side ofthis is a slightly more difficult check expression, that probably needs to dereference to a table.

Reply:
I'll resolve this comment, because even if it is agreed to validate against the HTTP field name syntax, this is a horrid way to specify it!